pkg/remote/aws_ecr_scanner_test.go

```diff
@@ -125,3 +125,69 @@ func TestECRRepository(t *testing.T) {
 		})
 	}
 }
+
+func TestECRRepositoryPolicy(t *testing.T) {
+	tests := []struct {
+		test           string
+		mocks          func(*repository.MockECRRepository, *mocks.AlerterInterface)
+		assertExpected func(t *testing.T, got []*resource.Resource)
+		err            error
+	}{
+		{
+			test: "single repository policy",
+			mocks: func(client *repository.MockECRRepository, alerter *mocks.AlerterInterface) {
+				client.On("ListAllRepositories").Return([]*ecr.Repository{
+					{RepositoryName: awssdk.String("test_ecr_repo_policy")},
+				}, nil)
+				client.On("GetRepositoryPolicy", &ecr.Repository{
+					RepositoryName: awssdk.String("test_ecr_repo_policy"),
+				}).Return(&ecr.GetRepositoryPolicyOutput{
+					RegistryId:     awssdk.String("1"),
+					RepositoryName: awssdk.String("test_ecr_repo_policy"),
+				}, nil)
+			},
+			assertExpected: func(t *testing.T, got []*resource.Resource) {
+				assert.Len(t, got, 1)
+
+				assert.Equal(t, got[0].ResourceId(), "test_ecr_repo_policy")
+			},
+			err: nil,
+		},
```

Comment:
We should have more tests cases here to test the resource listing errors of the enumerator

Reply:
Yeah the error test would make more sense here. But @almog27 is going to make another PR to support deepmode so I think it's better not spending too much time on this test that will be rewritten with goldenfile